src/python/WMComponent/RucioInjector/RucioInjectorPoller.py

```diff
@@ -407,24 +408,37 @@ def insertContainerRules(self):
         # Create the subscription objects and add them to the list
         # The list takes care of the sorting internally
         for subInfo in unsubscribedDatasets:
-            rse = subInfo['site']
+            kwargs = dict(ask_approval=False, activity="Production Output",
+                          account=self.rucioAcct, grouping="ALL",
+                          comment="WMAgent automatic container rule", meta=self.metaData)
+            rse = subInfo['site'].replace("_MSS", "_Tape")
             container = subInfo['path']
             if not self._isContainerTierAllowed(container):
                 logging.debug("Component configured to skip container rule for: %s", container)
                 continue
             logging.info("Creating container rule for %s against RSE %s", container, rse)
 
             rseName = "%s_Test" % rse if self.testRSEs else rse
-            # ALL = replicates all files to the same RSE
-            resp = self.rucio.createReplicationRule(container, rseExpression="rse=%s" % rseName,
-                                                    account=self.rucioAcct, grouping="ALL",
-                                                    comment="WMAgent automatic container rule",
-                                                    meta=self.metaData)
+            try:
+                # ALL = replicates all files to the same RSE
+                resp = self.rucio.createReplicationRule(container,
+                                                        rseExpression="rse=%s" % rseName, **kwargs)
+            except WMRucioException as exc:
+                msg = "Failed to create container rule for (retrying with approval): %s" % container
+                logging.warning(msg)
+                kwargs["ask_approval"] = True
+                try:
+                    resp = self.rucio.createReplicationRule(container,
+                                                            rseExpression="rse=%s" % rseName, **kwargs)
+                except Exception as exc:
+                    msg = "Failed once again to create container rule for: %s" % container
+                    msg += "\nWill retry again in the next cycle. Error: %s" % str(exc)
+                    continue
             if resp:
                 logging.info("Container rule created for %s under rule id: %s", container, resp)
                 subscriptionsMade.append(subInfo['id'])
             else:
-                logging.error("Failed to create rule for block: %s", container)
+                logging.error("Failed to create rule for container: %s", container)
```

Comment:
Out of curiosity, how is non-access-denied error handling accomplished? Looks like we log and move on here?

Reply:
We don't get a rule id from upstream, which means the same container will be retried over and over, until it succeeds; or until someone spots this problem in the logs :) Eventually we need to push this notification to the logdb.

Reply:
And here is what we discussed in the past weeks, on this subject: https://github.com/dmwm/WMCore/issues/9621